File: backend/app/api/routes/v1/dto/post.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field

from app.api.routes.v1.dto.tag import TagDTO
from app.api.routes.v1.dto.user import UserDTO


class PostCreationDTO(BaseModel):
    title: str = Field(max_length=100)
    description: str = Field(max_length=400)
    cover: UUID | None = None
    content: str
    tags: list[TagDTO]
    published: bool


class PostDTO(BaseModel):
    id: UUID | None = None
    title: str = Field(max_length=100)
    description: str = Field(max_length=400)
    cover: UUID | None = None
    likes: int
    content: str
    published: bool
    archived: bool
    featured: bool
    created_at: datetime
    author: UserDTO
    tags: list[TagDTO]


class PostMutationDTO(BaseModel):
    id: UUID
    title: str = Field(max_length=100)
    description: str = Field(max_length=400)
    cover: UUID | None = None
    content: str
    published: bool
    archived: bool

File: frontend/src/lib/api/dto/post.ts
import type { Tag } from "./tag";
import type { User } from "./user";

export interface Post {
  id: string;
  title: string;
  description: string;
  cover: string; // UUID
  likes: number;
  content: string;
  published: boolean;
  archived: boolean;
  featured: boolean;
  created_at: string; // ISO 8601 timestamp
  author: User;
  tags: Tag[];
  series?: string | null; // UUID
}

export interface PostCreationDTO {
  title: string;
  description: string;
  cover?: string | null;
  content: string;
  published: boolean;
  tags: Tag[];
  series?: string | null; // UUID
}

export interface PostUpdateDTO {
  id: string;
  title: string;
  description: string;
  cover?: string | null;
  content: string;
  published: boolean;
  archived: boolean;
  featured: boolean;
  tag_ids: string[];
  series?: string | null; // UUID
};

export interface PostTranslationResult {
  title: string;
  description: string;
  content: string;
}

export type SupportedLanguages = 
  | "English" 
  | "French" 
  | "Chinese" 
  | "Japanese" 
  | "Spanish" 
  | "German";

export type LanguageOption = SupportedLanguages | "Original";
